README.md

```diff
@@ -0,0 +1,173 @@
+
+# Remote Job Executor
+
+## **Introduction**
+This document outlines the design approach for developing a remote job executor that provides functionalities such as job creation, monitoring, control, and secure data transfer. Our primary objective is to design a system that is secure, efficient, and user-friendly.
+
+## **Scope**
+
+In this design, we aim to implement the level 5 requirements. This means that we need to implement:
+
+1. A CLI for communicating with the API server
+2. A library that allows clients to create jobs with resource control and isolation and allows multiple clients to stream a job's output.
+3. A gRPC server with mTLS authentication and a simple authorization scheme.   
+
+
+## **Design Approach**
+
+### **1. Command-Line User Experience (CLI UX)**
+
+The CLI will allow users to start and stop a job, query a job's state and stream their output. Below are some example commands. For job ids we are using random 12 character IDs. We are not using sequential numerical ids to prevent malicious users from guessing job ids.   
+
+* **Starting a Job:**  
+  ```
+  $ cli start echo "Hello World" 
+  Started job with ID: 4d5c9fa91db4
+  ```
+
+* **Starting a Job with flags/options:**
+  ```
+  # Per convention -- indicates the end of command options
+  $ cli start -- bash -c "echo Hello World"
+  Started job with ID: a2f3d7865b4e
+  ```
+
+* **Stopping a Job:**  
+  ```
+  $ cli stop 4d5c9fa91db4
+  ```
+
+* **Query Job Status:**
+  ```
+  $ cli status 4d5c9fa91db4
+  running, completed, failed or terminated 
+  ```
+
+* **Viewing Job Output:**  
+  ```
+  $ cli log 4d5c9fa91db4
+  ```
+
+* **Streaming Job Ouput:**
+  ```
+  $ cli log -f 4d5c9fa91db4
```

Comment:
I would do away with the `-f` and just make that the default behavior.

Reply:
Makes sense.